src/test/java/net/snowflake/ingest/streaming/internal/ParquetBufferTest.java

```diff
@@ -0,0 +1,278 @@
+package net.snowflake.ingest.streaming.internal;
+
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class ParquetBufferTest {
+  private ParquetRowBuffer rowBufferOnErrorContinue;
+
+  @Before
+  public void setupRowBuffer() {
+    this.rowBufferOnErrorContinue = createTestBuffer(OpenChannelRequest.OnErrorOption.CONTINUE);
+    this.rowBufferOnErrorContinue.setupSchema(RowBufferTest.createSchema());
+  }
+
+  ParquetRowBuffer createTestBuffer(OpenChannelRequest.OnErrorOption onErrorOption) {
+    SnowflakeStreamingIngestClientInternal<ParquetChunkData> client =
+        new SnowflakeStreamingIngestClientInternal<>("client");
+    SnowflakeStreamingIngestChannelInternal<ParquetChunkData> channel =
+        new SnowflakeStreamingIngestChannelInternal<>(
```

Comment:
we should consider a test builder for `SnowflakeStreamingIngestChannelInternal` as well at some point :)